ranch/storage.go

```diff
@@ -140,7 +141,21 @@ func (s *Storage) DeleteDynamicResourceLifeCycle(name string) error {
 			Namespace: s.namespace,
 		},
 	}
-	return s.client.Delete(s.ctx, o)
+	if err := s.client.Delete(s.ctx, o); err != nil {
+		return err
+	}
+	if err := wait.Poll(100*time.Millisecond, 5*time.Second, func() (bool, error) {
```

Comment:
why do we need to do this? and where did the value of 5s come from?

Reply:
The client is cache-backed, which means that it takes a moment for our own changes to be visible for us. This is important here, because otherwise we will a moment later get the drlc, not see this change, exclude the now-static resources from the number of static resources created for this dlrc, note that we have too little, create them and then a moment later attempt do delete all of them.

Reply:
The 100ms and 5s are relatively arbitrary, under normal circumstances the cache will lag between 200ms to 2s behind.